src/main/java/com/impactupgrade/nucleus/service/segment/XeroAccountingPlatformService.java

```diff
@@ -425,30 +463,78 @@ protected String getExceptionDetails(Exception e) {
+//                    .country(crmContact.mailingAddress.country);
+//            contact.setAddresses(List.of(address));
+//        }
+//
+//        // TODO: make this part not-sfdc specific?
+//        // TODO: SUPPORTER_ID_FIELD_NAME is DR specific
+//        if (crmContact.crmRawObject instanceof SObject sObject) {
+//            String supporterId = (String) sObject.getField(SUPPORTER_ID_FIELD_NAME);
+//            contact.setAccountNumber(supporterId);
+//        }
+//
+//        // TODO: temp
+//        env.logJobInfo("contact {} {} {} {} {} {}", crmContact.id, contact.getFirstName(), contact.getLastName(), contact.getName(), contact.getEmailAddress(), contact.getAccountNumber());
+//
+//        return contact;
+//    }
+
     protected Contact toContact(CrmContact crmContact) {
         if (crmContact == null) {
             return null;
         }
+
         Contact contact = new Contact();
-        contact.setFirstName(crmContact.firstName);
-        contact.setLastName(crmContact.lastName);
-        contact.setName(crmContact.getFullName());
+        contact.setAccountNumber(getAccountNumber(crmContact));
         contact.setEmailAddress(crmContact.email);
-        if (!Strings.isNullOrEmpty(crmContact.mailingAddress.street)) {
-            Address address = new Address()
-                    .addressLine1(crmContact.mailingAddress.street)
-                    .city(crmContact.mailingAddress.city)
-                    .region(crmContact.mailingAddress.state)
-                    .postalCode(crmContact.mailingAddress.postalCode)
-                    .country(crmContact.mailingAddress.country);
-            contact.setAddresses(List.of(address));
+
+        contact.setPhones(new ArrayList<>());
+        Phone mobilePhone = new Phone();
+        mobilePhone.setPhoneType(Phone.PhoneTypeEnum.MOBILE);
+        mobilePhone.setPhoneNumber(crmContact.mobilePhone);
+        //TODO: area/country codes?
+        contact.getPhones().add(mobilePhone);
+        if (!Strings.isNullOrEmpty(crmContact.workPhone)) {
+            Phone workPhone = new Phone();
+            workPhone.setPhoneType(Phone.PhoneTypeEnum.OFFICE);
+            workPhone.setPhoneNumber(crmContact.workPhone);
+            //TODO: area/country codes?
+            contact.getPhones().add(workPhone);
         }
 
-        // TODO: make this part not-sfdc specific?
-        // TODO: SUPPORTER_ID_FIELD_NAME is DR specific
-        if (crmContact.crmRawObject instanceof SObject sObject) {
-            String supporterId = (String) sObject.getField(SUPPORTER_ID_FIELD_NAME);
-            contact.setAccountNumber(supporterId);
+        if (crmContact.account.billingAddress != null) {
+            contact.setAddresses(List.of(toAddress(crmContact.account.billingAddress)));
+        }
+
+        if (crmContact.account.recordType == EnvironmentConfig.AccountType.HOUSEHOLD) {
+            // Household
+            contact.setName(crmContact.getFullName() + " " + contact.getAccountNumber());
```

Comment:
Suggestion: `contact.setName(crmContact.getFullName() + " " + getAccountNumber(crmContact));`. Use SFDC supporterId for household contact names and SFDC accountId for org names? SFDC IDs instead of the Contact account number? This would also make the lookup and mappings easier when we're sycing the donations?

Reply:
I'm a bit confused about business rules here, to be honest. 
We need to fill accountNumber and generate a unique contact name suffix.

Should these 2 pieces be the _same_ info? 
accountNumber = nameSuffix? (ex. accNum = 1; name = name + accNum) 
According to the doc - it seems to me that they should be the same (but correct me if I am wrong).

or we should have _separate_ pieces for accNum and nameSuffix and they should be _different_ values?
accountNumber = contact.supporterId (SFDC custom field) for households / contact.account.id (SFDC id) for orgs
and 
nameSuffix = contact.id (SFDC id) for households / contact.account.id (SFDC id) for orgs?